src/modules/75-cf/components/FlagArchiving/useArchiveFlagDialog.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useState } from 'react'
+import { Layout, Text, TextInput, useConfirmationDialog } from '@harness/uicore'
+import { Intent } from '@harness/design-system'
+import type { MutateRequestOptions } from 'restful-react/dist/Mutate'
+import { useToaster } from '@common/exports'
+import { String, useStrings } from 'framework/strings'
+import type { DeleteFeatureFlagQueryParams, Feature } from 'services/cf'
+import useResponseError from '@cf/hooks/useResponseError'
+import ArchiveFlagButtons from './ArchiveFlagButtons'
+
+export interface ArchiveDialogProps {
+  flagData: Feature
+  refetchFlags: () => void
+  deleteFeatureFlag: (
+    data: string,
+    mutateRequestOptions?: MutateRequestOptions<DeleteFeatureFlagQueryParams, void> | undefined
+  ) => void
+  queryParams: DeleteFeatureFlagQueryParams
+}
+
+interface UseArchiveFlagDialogReturn {
+  openDialog: () => void
+}
+
+const useArchiveFlagDialog = ({
+  flagData,
+  deleteFeatureFlag,
+  queryParams,
+  refetchFlags
+}: ArchiveDialogProps): UseArchiveFlagDialogReturn => {
+  const [isANameMatch, setIsANameMatch] = useState<boolean>(false)
```

Comment:
```suggestion
  const [isIdMatch, setIsIdMatch] = useState<boolean>(false)
```
should this be `isANameMatch ` as we are actually asking the user to type ID?